pubsub.go

```diff
@@ -326,60 +466,50 @@ func (p *PubsubValueStore) notifyWatchers(key string, data []byte) {
 	}
 }
 
-// rendezvous with peers in the name topic through provider records
-// Note: rendezvous/boostrap should really be handled by the pubsub implementation itself!
-func bootstrapPubsub(ctx context.Context, cr routing.ContentRouting, host host.Host, name string) {
-	// TODO: consider changing this to `pubsub:...`
-	topic := "floodsub:" + name
-	hash := u.Hash([]byte(topic))
-	rz := cid.NewCidV1(cid.Raw, hash)
+const sizeLengthBytes = 8
 
-	go func() {
-		err := cr.Provide(ctx, rz, true)
-		if err != nil {
-			log.Warningf("bootstrapPubsub: error providing rendezvous for %s: %s", topic, err.Error())
-		}
+// readNumBytesFromReader reads a specific number of bytes from a Reader, or returns an error
+func readNumBytesFromReader(r io.Reader, numBytes uint64) ([]byte, error) {
```

Comment:
This whole section feels really awkward to be here. Do we have a utility library that does this for us? If not maybe it's worth abstracting out the similar sections in go-libp2p-pubsub and here into something that can be used in general. On the other hand, it's really not much code so maybe we don't care.

Reply:
We usually use gogo's io library to read and decode a varint delimited stream.